core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java

```diff
@@ -99,74 +104,82 @@
 
   BypassMergeSortShuffleWriter(
       BlockManager blockManager,
-      IndexShuffleBlockResolver shuffleBlockResolver,
       BypassMergeSortShuffleHandle<K, V> handle,
       int mapId,
+      long mapTaskAttemptId,
```

Comment:
This is a new parameter, I remember there is interesting discussion around this in SPIP doc, about why this should be explicitly concerned in the new API, could we explain this? Maybe in `createMapOutputWriter`?

Reply:
Added javadoc